actors/market/tests/market_actor_test.rs

```diff
@@ -591,6 +591,331 @@ fn publish_a_deal_after_activating_a_previous_deal_which_has_a_start_epoch_far_i
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        provider_collateral,
+        client_collateral,
+        start_epoch,
+        end_epoch,
+    );
+    let qa_power = StoragePower::from_i128(1 << 50).unwrap();
+    rt.set_circulating_supply(qa_power); // convenient for these two numbers to cancel out
+
+    // publish the deal successfully
+    rt.set_epoch(publish_epoch);
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, worker_addr);
+    publish_deals(&mut rt, provider_addr, owner_addr, worker_addr, control_addr, &[deal]);
+    // TODO: actor.checkState(rt)
+}
+
+#[test]
+fn publish_multiple_deals_for_different_clients_and_ensure_balances_are_correct() {
+    let start_epoch = 42;
+    let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+
+    let mut rt = setup();
+
+    let client1_addr = Address::new_id(900);
+    let client2_addr = Address::new_id(901);
+    let client3_addr = Address::new_id(902);
+
+    let owner_addr = Address::new_id(OWNER_ID);
+    let provider_addr = Address::new_id(PROVIDER_ID);
+    let worker_addr = Address::new_id(WORKER_ID);
+    let control_addr = Address::new_id(CONTROL_ID);
+
+    // generate first deal for
+    let deal1 = generate_deal_and_add_funds(
+        &mut rt,
+        client1_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        start_epoch,
+        end_epoch,
+    );
+
+    // generate second deal
+    let deal2 = generate_deal_and_add_funds(
+        &mut rt,
+        client2_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        start_epoch,
+        end_epoch,
+    );
+
+    // generate third deal
+    let deal3 = generate_deal_and_add_funds(
+        &mut rt,
+        client3_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        start_epoch,
+        end_epoch,
+    );
+
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, worker_addr);
+    publish_deals(
+        &mut rt,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        control_addr,
+        &[deal1.clone(), deal2.clone(), deal3.clone()],
+    );
+
+    // assert locked balance for all clients and provider
+    let provider_locked =
+        &deal1.provider_collateral + &deal2.provider_collateral + &deal3.provider_collateral;
+    let client1_locked = get_locked_balance(&mut rt, client1_addr);
+    let client2_locked = get_locked_balance(&mut rt, client2_addr);
+    let client3_locked = get_locked_balance(&mut rt, client3_addr);
+    assert_eq!(deal1.client_balance_requirement(), client1_locked);
+    assert_eq!(deal2.client_balance_requirement(), client2_locked);
+    assert_eq!(deal3.client_balance_requirement(), client3_locked);
+    assert_eq!(provider_locked, get_locked_balance(&mut rt, provider_addr));
+
+    // assert locked funds dealStates
+    let st: State = rt.get_state();
+    let total_client_collateral_locked =
+        &deal1.provider_collateral + &deal2.provider_collateral + &deal3.provider_collateral;
```

Comment:
Not that it changes the test outcome given how generate_deal_proposal is written but total client collateral should be the sum of client collateral not provider collateral